gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiServiceImpl.java

```diff
@@ -681,6 +689,99 @@ public Page<GenericApiEntity> findAll(
             );
     }
 
+    @Override
+    public boolean isSynchronized(ExecutionContext executionContext, GenericApiEntity genericApiEntity) {
+        try {
+            // The state of the api is managed by kubernetes. There is no synchronization allowed from management.
+            if (DefinitionContext.isKubernetes(genericApiEntity.getDefinitionContext())) {
+                return true;
+            }
+
+            // 1 - Check if the api definition is sync with last one in events
+            Map<String, Object> properties = new HashMap<>();
+            properties.put(Event.EventProperties.API_ID.getValue(), genericApiEntity.getId());
```

Comment:
FYI, since Java11 you can use `Map.of((Event.EventProperties.API_ID.getValue(), genericApiEntity.getId())` to create a map with content in one line ;)